tests/components/marytts/test_tts.py

```diff
@@ -46,60 +49,83 @@ def test_setup_component(self):
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-    def test_service_say(self, aioclient_mock):
+    def test_service_say(self):
         """Test service call say."""
         calls = mock_service(self.hass, DOMAIN_MP, SERVICE_PLAY_MEDIA)
 
-        aioclient_mock.get(self.url, params=self.url_param, status=200, content=b"test")
+        conn = Mock()
+        response = Mock()
+        conn.getresponse.return_value = response
+        response.status = 200
+        response.read.return_value = b"audio"
 
         config = {tts.DOMAIN: {"platform": "marytts"}}
 
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        self.hass.services.call(
-            tts.DOMAIN, "marytts_say", {tts.ATTR_MESSAGE: "HomeAssistant"}
-        )
+        with patch("http.client.HTTPConnection", return_value=conn):
+            self.hass.services.call(
+                tts.DOMAIN, "marytts_say", {tts.ATTR_MESSAGE: "HomeAssistant"}
+            )
         self.hass.block_till_done()
 
-        assert len(aioclient_mock.mock_calls) == 1
         assert len(calls) == 1
+        assert calls[0].data[ATTR_MEDIA_CONTENT_ID].find(".wav") != -1
+        conn.request.assert_called_with("POST", "/process", urlencode(self.params))
 
-    def test_service_say_timeout(self, aioclient_mock):
-        """Test service call say."""
+    def test_service_say_with_effect(self):
+        """Test service call say with effects."""
         calls = mock_service(self.hass, DOMAIN_MP, SERVICE_PLAY_MEDIA)
 
-        aioclient_mock.get(
-            self.url, params=self.url_param, status=200, exc=asyncio.TimeoutError()
-        )
+        conn = Mock()
+        response = Mock()
+        conn.getresponse.return_value = response
+        response.status = 200
+        response.read.return_value = b"audio"
 
-        config = {tts.DOMAIN: {"platform": "marytts"}}
+        config = {
+            tts.DOMAIN: {"platform": "marytts", "effect": {"Volume": "amount:2.0;"}}
+        }
 
         with assert_setup_component(1, tts.DOMAIN):
             setup_component(self.hass, tts.DOMAIN, config)
 
-        self.hass.services.call(
-            tts.DOMAIN, "marytts_say", {tts.ATTR_MESSAGE: "HomeAssistant"}
-        )
+        with patch("http.client.HTTPConnection", return_value=conn):
+            self.hass.services.call(
+                tts.DOMAIN, "marytts_say", {tts.ATTR_MESSAGE: "HomeAssistant"}
+            )
         self.hass.block_till_done()
 
-        assert len(calls) == 0
-        assert len(aioclient_mock.mock_calls) == 1
+        assert len(calls) == 1
+        assert calls[0].data[ATTR_MEDIA_CONTENT_ID].find(".wav") != -1
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4878122)